hazelcast/src/main/java/com/hazelcast/map/impl/client/MapRemoveRequest.java

```diff
@@ -80,7 +79,7 @@ protected void beforeResponse() {
     }
 
     protected Operation prepareOperation() {
-        RemoveOperation op = new RemoveOperation(name, key);
+        MapOperation op = getOperationProvider().createRemoveOperation(name, key);
```

Comment:
why is the threadId not part of the arguments? Because then you can replace this code by:

protected Operation prepareOperation() {
   return getOperationProvider().createRemoveOperation(name, key, threadId);
}